packages/expo-cli/src/commands/client/index.js

```diff
@@ -249,132 +250,84 @@ export default program => {
-          });
-        if (sdkVersionStringOptions.length === 0) {
-          const answer = await prompt({
-            type: 'confirm',
-            name: 'updateToAClient',
-            message:
-              "It looks like we don't have a compatible client for your project. Do you want to try the latest client?",
-          });
+      const availableClients = ClientUpgradeUtils.getAvailableClients({
+        sdkVersions,
+        project: currentSdkConfig,
+        platform: 'ios',
+      });
 
-          if (answer.updateToAClient) {
-            await Simulator.upgradeExpoAsync();
-            log('Done!');
-            return;
-          }
+      if (availableClients.length === 0) {
+        const answer = await prompt({
+          type: 'confirm',
+          name: 'updateToAClient',
+          message: currentSdk
+            ? `We don't have a compatible client for SDK ${currentSdkVersion}. Do you want to try the latest client?`
+            : "It looks like we don't have a compatible client. Do you want to try the latest client?",
+        });
+        if (answer.updateToAClient) {
+          await Simulator.upgradeExpoAsync();
+          log('Done!');
         } else {
-          const answer = await prompt({
-            type: 'list',
-            name: 'selectedSdkVersionString',
-            message: 'Choose an SDK version to install the client for:',
-            pageSize: 20,
-            choices: sdkVersionStringOptions.map(optionSdkVersionString => {
-              const optionClientVersion = getIosVersion(sdkVersions[optionSdkVersionString]);
-              const message =
-                optionSdkVersionString === latestSdk.version
-                  ? `${chalk.bold(optionSdkVersionString)} ${chalk.grey(
-                      `- client ${optionClientVersion} (latest)`
-                    )}`
-                  : `${chalk.bold(optionSdkVersionString)} ${chalk.grey(
-                      `- client ${optionClientVersion}`
-                    )}`;
-
-              return {
-                value: optionSdkVersionString,
-                name: message,
-              };
-            }),
-          });
-
-          targetSdkVersion = sdkVersions[answer.selectedSdkVersionString];
-          targetSdkVersionString = answer.selectedSdkVersionString;
+          log('No client to install');
         }
-      }
-
-      if (!targetSdkVersion) {
-        log('No client to install.');
         return;
       }
 
-      if (await Simulator.upgradeExpoAsync(targetSdkVersion.iosClientUrl)) {
-        log(
-          `Done! We installed the client for SDK ${targetSdkVersionString}, version ${getIosVersion(
-            targetSdkVersion
-          )}`
-        );
+      const { targetClientUrl } = await prompt({
+        type: 'list',
+        name: 'targetClientUrl',
+        message: 'Choose an SDK version to install the client for:',
+        pageSize: 20,
+        choices: availableClients.map(client => {
```

Comment:
I think it would be good if we extract this too, although it's just asking questions it contains some logic on what to present to the developer.

Reply:
Did it anyway, the logic is definitely reusable.